src/File.cpp

```diff
@@ -7,30 +7,23 @@
 #include "chemfiles/files/GzFile.hpp"
 #include "chemfiles/files/XzFile.hpp"
 #include "chemfiles/ErrorFmt.hpp"
+#include "chemfiles/unreachable.hpp"
 using namespace chemfiles;
 
-static std::string extension(const std::string& filename) {
-    auto idx = filename.rfind('.');
-    if (idx != std::string::npos) {
-        return filename.substr(idx);
-    } else {
-        return "";
+std::unique_ptr<TextFile> TextFile::open(std::string path, File::Mode mode, File::Compression compression) {
+    switch (compression) {
+    case File::DEFAULT:
+        return std::unique_ptr<TextFile>(new PlainFile(std::move(path), mode));
```

Comment:
Is it to use move semantics? Does this offer performance improvements?

Reply:
More like semantic improvement: this is passed all the way down to `File`, which takes ownership of the string.

Reply:
Gotcha